Makefile.am

```diff
@@ -2,6 +2,10 @@ AUTOMAKE_OPTIONS=foreign
 
 AM_CFLAGS=-pedantic -Wall -Werror -Wextra
 
+if LINUX
```

Comment:
Replying to all comments above here. Unfortunately no, defining it on
*BSD makes BSD-specific functions like `pledge` and `reallocarray`
unavailable.

Maybe @mike-burns could shed some insight on how-to solve this?

The OS detection is taken from here [1] [2].

[1]: https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Canonicalizing.html
[2]: http://www.clearchain.com/blog/posts/autotools